types/module/module.go

```diff
@@ -110,8 +110,9 @@ func (bm BasicManager) ValidateGenesis(cdc codec.JSONMarshaler, txEncCfg client.
 
 // RegisterRESTRoutes registers all module rest routes
 func (bm BasicManager) RegisterRESTRoutes(clientCtx client.Context, rtr *mux.Router) {
+	r := rest.WithHTTPDeprecationHeaders(rtr)
```

Comment:
Actually shouldn't be be doing this module by module? Putting this in `BasicManager` means that other projects with other modules will have their REST routes auto-deprecated. That's not really what we want right?

Reply:
I thought this was what we wanted, as this entire process for generating REST routes will be deprecated at the SDK level in future releases, no?

Reply:
You're probably right... That would make SDK maintenance easier. Still this deprecation goes to clients. `RegisterRESTRoutes` should be deprecated for module developers. Module developers may decide not to deprecate their own REST routes and instead do that wiring on their own.

Reply:
Sorry- i'm not following exactly what you mean. Is this still something you are wanting changed? Or do you now think the current implementation is correct in the context of this PR?

Reply:
Yes. You're right in general but it still should be their decision. So I would prefer we do module by module.

Reply:
> REST routes for clients and instead do that wiring on their own.

This should be on by default to let the developers and users to know and prepare for that. 
At some point we will remove the REST API.

Reply:
If we need to coustomze it at the SDK level, then we can add a parameter to the `RegisterRESTRoutes` function:
```
 RegisterRESTRoutes(clientCtx client.Context, rtr *mux.Router, deprecateREST bool)
```